pkg/agent/agent.go

```diff
@@ -347,6 +357,20 @@ func (k *Klocksmith) getPodsForDeletion() ([]v1.Pod, error) {
 	return pods, nil
 }
 
+// waitForPodDeletion waits for a pod to be deleted
+func (k *Klocksmith) waitForPodDeletion(pod v1.Pod) error {
+	return wait.PollInfinite(time.Second, func() (bool, error) {
```

Comment:
Any opposition to making the reap timeout configurable? 10 minutes seems reasonable for most use cases but there are some that require a much longer timeout. Specifically, some of our pods running in our cluster can take up to 24 hours to terminate gracefully in certain scenarios.

Reply:
I'm not greatly in favor of new flags, unless really needed. Apps that require 24 hours to terminate gracefully, on clusters designed around the idea that individual nodes can go down and workloads move flexibly?

Reply:
If you've been using the original PR on clusters (and we mentioned that on any transient call error, the wait would exit), it sounds like your apps are already subject to ungraceful shutdown?

Reply:
We haven't been using it on our clusters for the very reason this PR was opened. The fact that pods don't terminate gracefully before reboot showed up in our testing prior to rolling this out.

Yep, it's most definitely not ideal that these jobs can take so long (most of the time they don't) and it's something that is being worked on but we have to work around it for the time being. 

24 hours is most definitely an outlier but I'd argue that 11 minutes isn't. 15 minutes still sounds pretty reasonable to me depending on what's running in the cluster. The point I'm trying to make is that the reap timeout is heavily dependent on the type of workload that runs on your cluster. Because it could be so variable, I'd say it's worth making into a flag.

Reply:
Ah sorry, misunderstood that last question. We’re used this PR on our test cluster running a similar workload but with not nearly as long of running.

Reply:
Hm, I'm not excited to add config flags (flags cascade into higher-level components), but I can see a `--grace-period` flag on update-agent (mirroring the --grace-period flag of `kubectl drain`) being warranted here. It does also set expectations that there has to be some period of time beyond which we give up on waiting (i.e. its best-effort, we can't guarantee we'll satisfy every pod grace period).